src/Parser.cpp

```diff
@@ -343,34 +343,45 @@ void Parser::parseString() {
                 v = (v << 4) + i - 'A' + 10;
               } else {
                 throw Exception(Exception::ParseError,
-                                "Illegal \\uXXXX escape sequence");
+                                "Illegal \\uXXXX escape sequence character");
               }
             }
             if (v < 0x80) {
               _builderPtr->appendByte(static_cast<uint8_t>(v));
-              highSurrogate = 0;
             } else if (v < 0x800) {
               _builderPtr->reserve(2);
               _builderPtr->appendByteUnchecked(0xc0 + (v >> 6));
               _builderPtr->appendByteUnchecked(0x80 + (v & 0x3f));
-              highSurrogate = 0;
-            } else if (v >= 0xdc00 && v < 0xe000 && highSurrogate != 0) {
-              // Low surrogate, put the two together:
-              v = 0x10000 + ((highSurrogate - 0xd800) << 10) + v - 0xdc00;
-              _builderPtr->rollback(3);
-              _builderPtr->reserve(4);
-              _builderPtr->appendByteUnchecked(0xf0 + (v >> 18));
-              _builderPtr->appendByteUnchecked(0x80 + ((v >> 12) & 0x3f));
-              _builderPtr->appendByteUnchecked(0x80 + ((v >> 6) & 0x3f));
-              _builderPtr->appendByteUnchecked(0x80 + (v & 0x3f));
-              highSurrogate = 0;
-            } else {
-              if (v >= 0xd800 && v < 0xdc00) {
+            } else if (v >= 0xdc00 && v < 0xe000) {
```

Comment:
Why not v <= 0xdfff?

Reply:
Does it make a difference? What is gained by changing `v >= 0xdc00 && v < 0xe000` to `v >= 0xdc00 && v <= 0xdffff`?

Reply:
No...
I don't looked bad :)
I thought you wrote some other number